### PR TITLE
algebra: adding tests for `quat_mlt` function

### PR DESCRIPTION
**Pull request by PNieck**
JIRA: PP-71

## Description

This PR contains tests for `quat_mlt` function.

## Motivation and Context

More precise tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (list targets here). `armv7a9-zynq7000-qemu`

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by niewim19 — APPROVED** [review]
Tests proposed in this PR are good to be merged. 

**Just an idea:** 
I do not know about your plans, but maybe permutation tests would be good. Something like:
 - initialize four quaternions (may be local): `a`, `i`, `j`, `k` (or in other notation of choice) each with only one element equal to `1`.
 - check if their products are according to multiplication table of quaternion space versors

**Comment by PNieck** [comment]
> Tests proposed in this PR are good to be merged.
> 
> **Just an idea:** I do not know about your plans, but maybe permutation tests would be good. Something like:
> 
>     * initialize four quaternions (may be local): `a`, `i`, `j`, `k` (or in other notation of choice) each with only one element equal to `1`.
> 
>     * check if their products are according to multiplication table of quaternion space versors

I like this idea - I will add such a test

**Comment by PNieck** [comment]
> > Tests proposed in this PR are good to be merged.
> > **Just an idea:** I do not know about your plans, but maybe permutation tests would be good. Something like:
> > ```
> > * initialize four quaternions (may be local): `a`, `i`, `j`, `k` (or in other notation of choice) each with only one element equal to `1`.
> > 
> > * check if their products are according to multiplication table of quaternion space versors
> > ```
> 
> I like this idea - I will add such a test

Test is ready

File: algebra/tests/quat/various.c [hunk 1]
```diff
@@ -380,3 +384,141 @@ TEST_GROUP_RUNNER(group_quat_dif)
...
+{
+	quat_t qA = { .a = 1.0f, .i = 0.0f, .j = 0.0f, .k = 0.0f };
+	quat_t qI = { .a = 0.0f, .i = 1.0f, .j = 0.0f, .k = 0.0f };
+	quat_t qJ = { .a = 0.0f, .i = 0.0f, .j = 1.0f, .k = 0.0f };
+	quat_t qK = { .a = 0.0f, .i = 0.0f, .j = 0.0f, .k = 1.0f };
+
+	quat_t qMinusA = { .a = -1.0f, .i = 0.0f, .j = 0.0f, .k = 0.0f };
+	quat_t qMinusI = { .a = 0.0f, .i = -1.0f, .j = 0.0f, .k = 0.0f };
+	quat_t qMinusJ = { .a = 0.0f, .i = 0.0f, .j = -1.0f, .k = 0.0f };
+	quat_t qMinusK = { .a = 0.0f, .i = 0.0f, .j = 0.0f, .k = -1.0f };
```
Comment (niewim19) [inline algebra/tests/quat/various.c:417]: More compact and readthrough approach :)
```suggestion
	quat_t qA, qI, qJ, qK, nqA, nqI, nqJ, nqK, qZero = { 0 };
	
	qA = qI = qJ = qK = nqA = nqI = nqJ = nqK = qZero; /* zero all quaternions */
	qA.a = qI.i = qJ.j = qK.k = 1.0f;                  /* set all positive unitary quaternions */
	nqA.a = nqI.i = nqJ.j = nqK.k = -1.0f;             /* set all negative unitary quaternions */
	
```

**Review by niewim19 — DISMISSED** [review]
I proposed a little simpler approach. However its not a merge blocker.

File: algebra/tests/quat/various.c [hunk 2]
```diff
@@ -380,3 +384,138 @@ TEST_GROUP_RUNNER(group_quat_dif)
...
+TEST(group_quat_mlt, quat_mlt_quatMltTable)
+{
+	quat_t qA, qI, qJ, qK, nqA, nqI, nqJ, nqK, qZero = { 0 };
+
+	qA = qI = qJ = qK = nqA = nqI = nqJ = nqK = qZero; /* zero all quaternions */
+	qA.a = qI.i = qJ.j = qK.k = 1.0f;                  /* set all positive unitary quaternions */
+	nqA.a = nqI.i = nqJ.j = nqK.k = -1.0f;             /* set all negative unitary quaternions */
+
+
+	quat_t Res;
```
Comment (niewim19) [inline algebra/tests/quat/various.c:416]: Double newline + declaration goes first.
```suggestion
	quat_t qA, qI, qJ, qK, nqA, nqI, nqJ, nqK;
	quat_t Res, qZero = { 0 };

	qA = qI = qJ = qK = nqA = nqI = nqJ = nqK = qZero; /* zero all quaternions */
	qA.a = qI.i = qJ.j = qK.k = 1.0f;                  /* set all positive unitary quaternions */
	nqA.a = nqI.i = nqJ.j = nqK.k = -1.0f;             /* set all negative unitary quaternions */
```

**Review by niewim19 — APPROVED** [review]
LGTM

**Pull request merged**